src/server/SecureKeyValueStore.ts

```diff
@@ -1,23 +1,40 @@
 import {KeyValueStore} from "ferrum-plumbing";
+import {SecretHanlder} from './SecretsHandler';
+import {KmsCryptor} from "aws-lambda-helper";
 
 // TODO: Implement
 // You need to set up storage and provide relevant configurations. Firestore can be an option.
 // Bonus: encrypt / decrypt data using AWS KMS before storage
 export class SecureKeyValueStore implements KeyValueStore {
-    constructor() {
+    constructor(
+        private secretsHandler: SecretHanlder,
+        private cryptor: KmsCryptor,
+        ) {
     }
 
-    getItem<T>(k: string): Promise<T | undefined> {
-        // TODO: Get an item and decrypt
-        return undefined;
+    __name__(): string { return 'SecureKeyValueStore'; }
+
+    async getItem<T>(k: string): Promise<any> {
```

Comment:
Keep this method typed. `getItem<T>(k: string): Promise<T>` and cast the response to `T` at the end: `return response as T;`